src/auto-advance.js

```diff
@@ -0,0 +1,58 @@
+import window from 'global/window';
+
+/**
+ * Validates a number of seconds to use as the auto-advance delay.
+ *
+ * @param  {Number} s
+ * @return {Boolean}
+ */
+const validSeconds = s =>
+  typeof s === 'number' && !isNaN(s) && s >= 0 && s < Infinity;
+
+/**
+ * Resets the auto-advance behavior of a player.
+ *
+ * @param  {Player} player
+ */
+const reset = (player) => {
+  if (player.playlist.autoadvance_.timeout) {
+    window.clearTimeout(player.playlist.autoadvance_.timeout);
+  }
+
+  if (player.playlist.autoadvance_.trigger) {
+    player.off('ended', player.playlist.autoadvance_.trigger);
+  }
+
+  player.playlist.autoadvance_.timeout = null;
+  player.playlist.autoadvance_.trigger = null;
+};
+
+/**
+ * Sets up auto-advance behavior on a player.
+ *
+ * @param  {Player} player
+ * @param  {Number} seconds
+ */
+const setup = (player, seconds) => {
+  reset(player);
```

Comment:
This seems like behavior change. However, it seems like it keeps the same behavior but is a bit more simple.
We should document the behavior on the doc for the function. Something like:

auto-advance is setup up when called with a valid timeout in seconds. When called with a bogus value for the timeout or when called while a timeout is in affect, the auto-advance is cancelled and removed. Otherwise, set up a new auto-advance timeout with the newly provided time.